### PR TITLE
remove dependency on oops::config_mod

### LINKED ISSUE #171: Incorrect clone of geometry
https://github.com/JCSDA/soca/blob/8a253b64e593f0e80d23bebd27b3427240787de6/src/soca/Geometry/soca_geom_mod.F90#L132

Initializing does not make it a clone!

### PR DESCRIPTION
**Pull request by aerorahul**
This PR removes the dependency of OOPS::config_mod  See: https://github.com/JCSDA/oops/pull/330

- `c_conf` is no longer in any of the fortran modules, except in the interface.  The interface passes the fckit configuration down to the fortran code.

Some clean-up is performed along the way, added explicitly what is used from the fortran modules e.g. `use kinds, only: kind_real` etc.

Also closes #171

### CODE REVIEW

**Comment by travissluka** [comment]
The dirac tests are failing for me when it tries to read in the yaml file. 
```
The following tests FAILED:
         22 - test_soca_dirac_soca_cov (Failed)
         23 - test_soca_dirac_socahyb_cov (Failed)

22: Exception: Bad Conversion: Cannot convert (1,17,41,31,51,63,81,14,16,43) (List) to long long  (/home/tsluka/work/jedi-soca/soca.src/eckit/src/eckit/value/Content.cc +74 badConvertion)
22: Exception: Bad Conversion: Cannot convert (1,17,41,31,51,63,81,14,16,43) (List) to long long  (/home/tsluka/work/jedi-soca/soca.src/eckit/src/eckit/value/Content.cc +74 badConvertion)
22: Bad Conversion: Cannot convert (1,17,41,31,51,63,81,14,16,43) (List) to long long caught in  (/home/tsluka/work/jedi-soca/soca.src/oops/src/oops/runs/Run.cc +71 execute)
22: Exception: oops::Dirac<SOCA> terminating... 
```

File: src/soca/Fields/soca_fields_mod.F90 [hunk 1]
```diff
@@ -242,11 +254,19 @@ subroutine dirac(self, c_conf)
     f_comm = fckit_mpi_comm()
 
     ! Get Diracs size
-    ndir = config_get_data_dimension(c_conf,'ixdir')
-    if ((config_get_data_dimension(c_conf,'iydir')/=ndir) .or. &
-         (config_get_data_dimension(c_conf,'izdir')/=ndir) .or. &
-         (config_get_data_dimension(c_conf,'ifdir')/=ndir)) &
-         & call abor1_ftn('qg_fields_dirac: inconsistent sizes for ixdir, iydir, izdir, ipdir and ifdir')
+    if ( f_conf%has("ixdir") ) &
+        call f_conf%get_or_die("ixdir", ndir)
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
        ndir = f_conf%get_size("ixdir")
```

File: src/soca/Fields/soca_fields_mod.F90 [hunk 2]
```diff
@@ -242,11 +254,19 @@ subroutine dirac(self, c_conf)
...
+    if ( f_conf%has("iydir") ) &
+        call f_conf%get_or_die("iydir", niydir)
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
        niydir = f_conf%get_size("iydir")
```

File: src/soca/Fields/soca_fields_mod.F90 [hunk 3]
```diff
@@ -242,11 +254,19 @@ subroutine dirac(self, c_conf)
...
+    if ( f_conf%has("izdir") ) &
+        call f_conf%get_or_die("izdir", nizdir)
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
        nizdir = f_conf%get_size("izdir")
```

File: src/soca/Fields/soca_fields_mod.F90 [hunk 4]
```diff
@@ -242,11 +254,19 @@ subroutine dirac(self, c_conf)
...
+    if ( f_conf%has("ifdir") ) &
+        call f_conf%get_or_die("ifdir", nifdir)
```
Comment (travissluka) [inline src/soca/Fields/soca_fields_mod.F90]:
```suggestion
        nifdir = f_conf%get_size("ifdir")
```

**Comment by aerorahul** [comment]
@travissluka thanks for the catch. The bug has been removed.

**Review by travissluka — APPROVED** [review]
great, seems good to go once https://github.com/JCSDA/oops/pull/330 is merged

**Comment by guillaumevernieres** [comment]
@aerorahul merging soca #173 with your branch worked for me, I am not seeing the ctest failures that you were describing to me offline, but I haven't tested with intel yet.

**Comment by aerorahul** [comment]
@travissluka @guillaumevernieres
I say we merge this branch into develop and use 
```
oops::feature/remove_config_mod
ufo::feature/remove_config_mod
```

**Comment by guillaumevernieres** [comment]
Sounds good to me. @travissluka ?

**Comment by travissluka** [comment]
eh, since we don't urgently need this to fix something broken, I would prefer to stay on the develop upstream repos. I'm waiting on an oops PR to be merged into develop that's needed to fix issues with our ctests https://github.com/JCSDA/oops/pull/341

**Comment by aerorahul** [comment]
@travissluka fair enough. However, I don’t want to create a backlog on soca development, that is building on this work.
Do we have a estimate on when those PR’s will make it to develop branches?

**Comment by guillaumevernieres** [comment]
> @travissluka fair enough. However, I don’t want to create a backlog on soca development, that is building on this work.
> Do we have a estimate on when those PR’s will make it to develop branches?

I'm with @aerorahul , but I'm not planning to push anything until late next week, so I'm OK waiting. We can re-assess next week, and unlike what we usually do, lets (try to) merge the older pr first!
@aerorahul , is your cleanup branch based off of this branch?

**Comment by aerorahul** [comment]
@guillaumevernieres Yes. The cleanup branch is branched from this branch. 
At the very least, once that branch is reviewed and approved, I would like it to be merged into this one, if that is OK with @travissluka, so I can continue along other work.
Does that sound like a plan?

**Comment by guillaumevernieres** [comment]
@aerorahul OK, my bad, did not see you were trying to merge into this branch. I'll have a look right now.

**Comment by guillaumevernieres** [comment]
@aerorahul : Is your next step to move the linked list to the interface level? If so, can you add an issue in zenhub, or update an existing issue?

**Comment by travissluka** [comment]
@aerorahul  @guillaumevernieres  Ah, yeah i didn't see either that the cleanup branch was to be merged into here. Everything's cool now!

**Comment by aerorahul** [comment]
> @aerorahul : Is your next step to move the linked list to the interface level? If so, can you add an issue in zenhub, or update an existing issue?

yes. The linkedLists are now all at the interface level.  It was done as part of #178.  I don't see an existing issue for this.

**Comment by travissluka** [comment]
oops, did I click too early?

**Comment by aerorahul** [comment]
> oops, did I click too early?

No, you are good.  Doing a test on develop now, and then will delete this branch.

**Pull request merged**